pkg/apis/certmanager/v1/types.go

```diff
@@ -30,6 +30,9 @@ const (
 	// Annotation key for certificate common name.
 	CommonNameAnnotationKey = "cert-manager.io/common-name"
 
+	// Annotation key for certificate renewBefore.
+	RenewBeforeAnnotationKey = "cert-manager.io/renew-before"
```

Comment:
I actually overlooked this... we have no duration annotation... so what if i set rebew-before to 100 days?

Reply:
it actually would cause an admission hook error...